core/store/src/trie/trie_storage.rs

```diff
@@ -59,6 +58,10 @@ pub trait TrieStorage {
     fn as_partial_storage(&self) -> Option<&TrieMemoryPartialStorage> {
         None
     }
+
+    fn get_touched_nodes_count(&self) -> u64 {
+        unimplemented!();
+    }
```

Comment:
```suggestion
    fn get_touched_nodes_count(&self) -> u64;
```

Reply:
@matklad why? This method doesn't make sense for impls except `TrieCachingStorage` and `TrieRecordingStorage`, so if we incorrectly call it for some other impl, I prefer to see general unimplemented error instead of implementing it for each `TrieStorage` impl separately.

Reply:
However, I filed an issue that this counter should work with recording storage as well https://github.com/near/nearcore/issues/6316

Reply:
No strong opinion here, it's just that `unimplemented!` as a default impl feels like a smell.

`unimplemented!()` in a trait impl is a smell -- if someone could be unimplemented, it's generally better to split a trait in two. I don't think we can easy make this change here (and I'd probably argue that an impl which returns 0 would be more correct), but we should make this `unimplemented!()` stand out more by explicitly writing this for every type where this method doesn't make sense.

Reply:
+1 to moving the 'unimplemented' part into each of the implementation classes.